kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -274,33 +313,103 @@ public String createConsumer(String group, ConsumerInstanceConfig instanceConfig
       return null;
     }
 
-    int workerId = nextWorker.getAndIncrement() % workers.size();
-    KafkaConsumerWorker worker = workers.get(workerId);
-    return worker.readRecords(
-        state, timeout, maxBytes,
-        new ConsumerWorkerReadCallback<ClientKeyT, ClientValueT>() {
-          @Override
-          public void onCompletion(
-              List<? extends ConsumerRecord<ClientKeyT, ClientValueT>> records, Exception e
-          ) {
-            updateExpiration(state);
-            if (e != null) {
-              // Ensure caught exceptions are converted to RestExceptions so the user gets a
-              // nice error message. Currently we don't define any more specific errors because
-              // the old consumer interface doesn't classify the errors well like the new
-              // consumer does. When the new consumer is available we may be able to update this
-              // to provide better feedback to the user.
-              Exception responseException = e;
-              if (!(e instanceof RestException)) {
-                responseException = Errors.kafkaErrorException(e);
+    final KafkaConsumerReadTask task = new KafkaConsumerReadTask<KafkaKeyT, KafkaValueT,
+          ClientKeyT, ClientValueT>(
+          state,
+          timeout,
+          maxBytes,
+          new ConsumerWorkerReadCallback<ClientKeyT, ClientValueT>() {
+            @Override
+            public void onCompletion(
+                    List<? extends ConsumerRecord<ClientKeyT, ClientValueT>> records,
+                    Exception e) {
+              KafkaConsumerManager.this.updateExpiration(state);
+              if (e != null) {
+                // Ensure caught exceptions are converted to RestExceptions so the user gets
+                // a nice error message. Currently we don't define any more specific
+                // errors because the old consumer interface doesn't classify the errors well
+                // like the new consumer does. When the new consumer is available we may be
+                // able to update this to provide better feedback to the user.
+                Exception responseException = e;
+                if (!(e instanceof RestException)) {
```

Comment:
It's a little annoying that we have to repeat this check since we already did it prior to invoking `onCompletion`. As far as I can tell, we should be able to use the more specific `RestException` type in the callback signature and avoid some of these redundant checks:

```java
  public interface ConsumerWorkerReadCallback <K, V> {
    void onCompletion(List<? extends ConsumerRecord<K, V>> records, RestException e);
  }
```

On a related note, the comment above mentions that we should use `RestException` to ensure the user gets a nice message. That makes sense to me, so perhaps we should do the same thing in `ConsumerReadTask.finish(Exception)`.